tests/plots/test_declarative.py

```diff
@@ -437,6 +437,43 @@ def test_declarative_events():
     return pc.figure
 
 
+@pytest.mark.mpl_image_compare(remove_text=True, tolerance=0)
+@needs_cartopy
+def test_declarative_raster_events():
+    """Test that resetting traitlets properly propagates in RasterPlot()."""
+    data = xr.open_dataset(get_test_data('narr_example.nc', as_file_obj=False))
+
+    raster = RasterPlot()
+    raster.data = data
+    raster.field = 'Geopotential_height'
+    raster.level = 850 * units.hPa
+
+    img = ImagePlot()
```

Comment:
I don't think we need the `ImagePlot()` as part of this test. We just want to change the `RasterPlot()` items.